source/elements/oneMKL/source/domains/spblas/data_types/create_coo_matrix.rst

```diff
@@ -0,0 +1,241 @@
+two sparse matrices, the number of non-zero elements in the output sparse matrix
+is not known in advance and must be calculated as part of the operation API.
+Such APIs are currently not a part of the current oneMKL Specification, but will
+be added in the future. Therefore, it is important for the ``create_coo_matrix``
+API to be able to handle multiple calls to build the sparse matrix as function
+arguments become known and arrays are allocated with the correct sizes. In
+particular, the ``create_coo_matrix`` API must handle the case where the number
+of non-zero elements in the matrix are unknown and the matrix arrays are
+``nullptr`` or zero-sized ``sycl::buffer`` objects as applicable. This behavior
+is currently left to be implementation-defined, but may be clarified in the
+oneMKL Specification in the future.
+
+See :ref:`onemkl_sparse_matrix_handle`.
+
+.. _onemkl_sparse_create_coo_matrix_buffer:
+
+create_coo_matrix (Buffer version)
+----------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void create_coo_matrix (sycl::queue                          &queue,
+                               oneapi::mkl::sparse::matrix_handle_t *p_smhandle,
+                               std::int64_t                         num_rows,
+                               std::int64_t                         num_cols,
+                               std::int64_t                         nnz,
+                               index_base                           index,
+                               sycl::buffer<indexType, 1>           &row_ind,
+                               sycl::buffer<indexType, 1>           &col_ind,
+                               sycl::buffer<dataType, 1>            &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+   indexType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_smhandle
+      The address of the ``p_smhandle`` object to be initialized. Must only be
+      called on an uninitialized ``matrix_handle_t`` object.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
+
+   nnz
+      Number of non-zero entries in the matrix (which may include explicit
+      zeros). Must be at least 0.
+
+   index
+      Indicates how input arrays are indexed. The possible options are described
+      in :ref:`onemkl_enum_index_base` enum class.
+
+   row_ind
+      Buffer of length ``nnz`` containing the row indices in ``index``-based
```

Comment:
Maybe this should be the following?
```suggestion
      Buffer of length at least ``nnz`` containing the row indices in ``index``-based
```

Reply:
Addressed in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/13dd2e5721d52c730695e51fe3b5471580f0d75e